File: src-tauri/src/ai/providers/qianhai/models/gemini_3_pro_image_preview.rs
use super::shared::GoogleCompatibleImagePreviewAdapter;

inventory::submit! {
    crate::ai::providers::qianhai::models::RegisteredQianhaiModel {
        build: || Box::new(GoogleCompatibleImagePreviewAdapter::new(
            "qianhai/gemini-3-pro-image-preview",
            &[
                "qianhai/gemini-3-pro-image-preview",
                "gemini-3-pro-image-preview",
            ],
        )),
    }
}

File: src/features/canvas/models/image/qianhai/gemini3ProImagePreview.ts
import { createQianhaiImagePreviewModel } from './shared';

export const QIANHAI_GEMINI_3_PRO_IMAGE_PREVIEW_MODEL_ID =
  'qianhai/gemini-3-pro-image-preview';

export const imageModel = createQianhaiImagePreviewModel({
  id: QIANHAI_GEMINI_3_PRO_IMAGE_PREVIEW_MODEL_ID,
  displayName: 'nano banana pro',
  description: '千海 · Gemini 3 Pro Image Preview 图像生成与编辑',
  requestModel: 'gemini-3-pro-image-preview',
});
